### ISSUE TITLE
5.10 & 5.11 break local development with Next.js

### ISSUE THREAD

**Issue by samtgarson** [prisma/prisma#23616]
### Bug description

I have a Next.js application connecting to a Neon database in production, and a local Postgres database in development.

I use NextAuth, which requires importing the Prisma client as part of the configuration, however I have it configured not to require connecting to a database in the edge middleware (using JWT sessions).

Up to 5.9, I have been able to run my application using the Neon serverless driver _only in production_, and no driver in local development without a problem.

After upgrading past 5.10, I get the following errors just initializing the Prisma client:
```
5.10.0
Error [PrismaClientValidationError]: PrismaClient failed to initialize because it wasn't configured to run in this environment (edge-light) 

5.11.0
Error [PrismaClientValidationError]: PrismaClient failed to initialize because it wasn't configured to run in this environment (Vercel Edge Functions or Edge Middleware).
```

My hunch (I could be wrong) is that this is because whereas previously you threw an error when trying to _connect_ to the database in an unsupported environment, you now check when _initializing_.

As node-postgres is not supported in the Edge environment, do you have any recommendations for how to develop an application against a local database with this change? 

(This isn't the first time I've had development against a local database broken, I'm wondering if I'm the only one not connecting to a hosted database during development...)

### How to reproduce

Set up a Next.js application and **import** the prisma client in an Edge runtime or Edge Middleware file.

### Expected behavior

I should be able to import the client without using it in an edge environment, even if I'm not using an edge driver I guess?

### Prisma information

```ts
import { Pool } from '@neondatabase/serverless'
import { PrismaNeon } from '@prisma/adapter-neon'
import { PrismaClient } from '@prisma/client'

let prisma: PrismaClient

if (process.env.NODE_ENV === 'production') {
  prisma = new PrismaClient({ adapter: adapter() })
} else {
  const globalWithPrisma = global as typeof globalThis & {
    prisma: PrismaClient
  }
  if (!globalWithPrisma.prisma) {
    globalWithPrisma.prisma = new PrismaClient()
  }
  prisma = globalWithPrisma.prisma
}

export default prisma
export * from '@prisma/client'

function adapter() {
  if (!process.env.VERCEL) return null

  const connectionString = `${process.env.DATABASE_URL}`
  const pool = new Pool({ connectionString })
  return new PrismaNeon(pool)
}
```

### Environment & setup

- OS: macOS
- Database: PostgreSQL locally, Neon in production
- Node.js version: 20.x

### Prisma Version

```
5.10.2 through 5.11.0
```

**Comment by SoraKumo001** [comment]
Due to this change, all Adapters are now required to be util.
https://github.com/prisma/prisma/pull/23013

**Comment by SevInf** [comment]
@SoraKumo001 could you elaborate? I am not sure what you mean by "required to be util" and in any case, requiring anything was not the intended change.

**Comment by SoraKumo001** [comment]
https://github.com/prisma/prisma/pull/23013/files#diff-6a6003f13f8c6e946d783879e01c29cde59abb856aa36ba3109ed6c989a23a1aR71

The above change requires node-runtime for all Adapters because Next.js webpack incorporates 'util' into the bundle.
Using Adapter with CloudflareWorkers also requires 'node_compat'.

**Comment by SevInf** [comment]
@SoraKumo001 I don't think this related to whatever OP reports. Could you open a separate issue with reporoduction repo? Thank you.
